### PR TITLE
[DOC] update doc for 24.12 release [skip ci]

### PR DESCRIPTION
**Pull request by nvliyuan**
This PR updates download docs for 24.12.0 release.
Signed-off-by: liyuan <yuali@nvidia.com>

### CODE REVIEW

File: docs/download.md [hunk 1]
```diff
@@ -85,24 +85,27 @@ CUDA 11.8 through CUDA 12.0.
...
 	gpg: Good signature from "NVIDIA Spark (For the signature of spark-rapids release jars) <sw-spark@nvidia.com>"
 
 ### Release Notes
-* Optimize scheduling policy for GPU Semaphore
-* Support distinct join for right outer joins
-* Support MinBy and MaxBy for non-float ordering  
-* Support ArrayJoin expression
-* Optimize Expand and Aggregate expression performance
-* Improve JSON related expressions
+* Implement Kudo shuffle format to improve shuffle performance
```
Comment (jlowe) [inline docs/download.md:97]: This disabled by default with an internal config.  There's no documentation on how to enable this, so I'm not sure it's appropriate to call it out here.  It also doesn't improve shuffle performance for small schemas and/or relatively small partition counts (e.g.: default 200 or less).

File: docs/download.md [hunk 2]
```diff
@@ -85,24 +85,27 @@ CUDA 11.8 through CUDA 12.0.
...
+* Add the ability to decompress snappy and ztsd compressed Parquet files via CPU
```
Comment (jlowe) [inline docs/download.md:98]: This is also an internal-only config with no documentation.  Not sure this is appropriate to advertise.

File: docs/download.md [hunk 3]
```diff
@@ -85,24 +85,27 @@ CUDA 11.8 through CUDA 12.0.
...
+* Add repartition-based algorithm fallback in hash aggregate  
+* Support spark function months_between
```
Comment (jlowe) [inline docs/download.md:100]:
```suggestion
* Support Spark function months_between
```

File: docs/download.md [hunk 4]
```diff
@@ -85,24 +85,27 @@ CUDA 11.8 through CUDA 12.0.
...
+* Support asynchronous writing for Parquet files
+* Add retry support to improve sub hash-join stability
+* Improve JSON scan and from_json
+* Support setting a limited number of tasks for build-in profiling tool
```
Comment (jlowe) [inline docs/download.md:104]: Same issue here, the self-profiling tool is still experimental and not publicly documented yet.

File: docs/download.md [hunk 5]
```diff
@@ -85,24 +85,27 @@ CUDA 11.8 through CUDA 12.0.
...
+* Bugs fix for memory leak
```
Comment (jlowe) [inline docs/download.md:105]: Ideally this should be a bit more descriptive.  Is it a host or device memory leak?  What are the conditions for the leak, i.e.: what types of operations were affected?

File: docs/download.md [hunk 6]
```diff
@@ -85,24 +85,27 @@ CUDA 11.8 through CUDA 12.0.
...
 * For updates on RAPIDS Accelerator Tools, please visit [this link](https://github.com/NVIDIA/spark-rapids-tools/releases)
```
Comment (jlowe) [inline docs/download.md:106]: There's no mention of:
- falling back to the CPU for ORC boolean writes by the GPU due to a bug in cudf's ORC writer
- Improved performance for CASE WHEN statements comparing a string column against multiple values

File: docs/download.md (hunk 5 above)
Comment (nvliyuan) [reply docs/download.md]: Double check the PRs that we don't have a memory leak bug fix this release, will remove it.

**Review by jlowe — DISMISSED** [review]

**Comment by abellina** [comment]
Just double checking that this PR can go in after https://github.com/NVIDIA/spark-rapids/pull/11848. Or, should it be pulled into the merge to main branch? Probably doesn't matter too much.

File: docs/download.md (hunk 5 above)
Comment (abellina) [reply docs/download.md]: We have at least this one: https://github.com/NVIDIA/spark-rapids/pull/11848/commits/3449c8a772899ff1752ea42efa89f1ddc049cc6a

Comment (abellina) [reply docs/download.md]: and this one https://github.com/NVIDIA/spark-rapids/pull/11848/commits/96a58d121a5af7ef956196a6141fe7777277e95e

Comment (nvliyuan) [reply docs/download.md]: updated, thx

**Comment by nvliyuan** [comment]
> Just double checking that this PR can go in after #11848. Or, should it be pulled into the merge to main branch? Probably doesn't matter too much.

this pr need to go in before #11848 as we used to do

**Comment by NvTimLiu** [comment]
> > Just double checking that this PR can go in after #11848. Or, should it be pulled into the merge to main branch? Probably doesn't matter too much.
> 
> this pr need to go in before #11848 as we used to do

Yes, we'll do the final up-merge for merge-to-main PR #11848, to include all PRs targeting to the branch-24.12, then merge them to main!

**Review by NvTimLiu — APPROVED** [review]
LGTM

**Comment by nvliyuan** [comment]
Hi @jlowe @abellina , can you please help check again, thx

**Review by jlowe — APPROVED** [review]

**Comment by nvliyuan** [comment]
build

**Pull request merged**